src/book/core.clj

```diff
@@ -0,0 +1,40 @@
+(ns book.core)
+(ns book.core
+  (:require [clojure.test :refer [is run-tests with-test]]))
+
+;TODO Write some tests
+
+(let [books '({:title "Romeo and Juliet" :author-id 890}
+              {:title "Moby Dick" :author-id 123}
+              {:title "The Idiot" :author-id 567}
+              {:title "Bartleby, the Scrivener" :author-id 123}
+              {:title "Crime and Punishment" :author-id 567}
+              {:title "Of Mice and Men" :author-id 248}
+              {:title "Harry Potter and the Chamber of Secrets" :author-id 135})
+
+      authors {123 "Herman Melville" 567 "Fyodor Dostoevsky"
+               890 "William Shakespeare" 135 "J.K. Rowling"}]
+  (def catalog {:books books :authors authors}))
+
+(with-test (defn search [in-title catalog]
```

Comment:
It's not very common to write tests inside the source code. Usually, we write tests in a separate namespace.